src/main/java/org/example/HardAI.java

```diff
@@ -0,0 +1,74 @@
+package org.example;
+
+import static tictactoe.Main.*;
+
+public class HardAI extends Player {
+    @Override
+    public char[][] play(char[][] matrix, char yourChar) {
+        System.out.println("Making move level \"hard\"");
+        int[] bestMove = getBestMove(matrix, yourChar);
+        matrix[bestMove[0]][bestMove[1]] = yourChar;
+        return matrix;
+    }
+
+    public int[] getBestMove(char[][] matrix, char yourChar) {
+        int[] bestMove = new int[]{-1, -1};
+        int bestScore = (yourChar == 'X') ? Integer.MIN_VALUE : Integer.MAX_VALUE;
+
+        for (int i = 0; i < matrix.length; i++) {
+            for (int j = 0; j < matrix[i].length; j++) {
+                if (matrix[i][j] == ' ') {
+                    matrix[i][j] = yourChar;
+                    int score = minimax(matrix, 10, otherChar(yourChar));
+                    matrix[i][j] = ' ';
+
+                    if ((yourChar == 'X' && score > bestScore) || (yourChar == 'O' && score < bestScore)) {
```

Comment:
you can also try to extract such long boolean expressions as separate methods with a perfect name. So that I don't even need to look into implementation and directly understand if statement with that name.